python/pyarrow/ipc.py

```diff
@@ -56,9 +56,16 @@ class RecordBatchStreamReader(lib._RecordBatchStreamReader, _ReadPandasOption):
     source : bytes/buffer-like, pyarrow.NativeFile, or file-like Python object
         Either an in-memory buffer, or a readable file object
     """
+
     def __init__(self, source):
         self._open(source)
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        pass
```

Comment:
Seems like we don't have a close method on the readers, but I've added it for convenience.